jina/executors/decorators.py

```diff
@@ -313,29 +313,30 @@ def _batching(func):
         def arg_wrapper(*args, **kwargs):
             # priority: decorator > class_attribute
             # by default data is in args[1:] (self needs to be taken into account)
-            data_args = [args[slice_on + i] for i in range(0, num_data)]
-
-            data = list(zip(*data_args))
-            args = list(args)
-
             b_size = batch_size or getattr(args[0], 'batch_size', None)
             # no batching if b_size is None
             if b_size is None:
                 return func(*args, **kwargs)
 
+            data_args = [args[slice_on + i] for i in range(0, num_data)]
+            args = list(args)
```

Comment:
Why that? I always thought `*args` must be a list.

Reply:
it seems it is a tuple

Reply:
Yes, it is a tuple and we pass it to a list to make it mutable

Reply:
So it is something done already in the previous `batching`

Reply:
Cool interesting insight. Thanks for clarification